manifests/install.pp

```diff
@@ -310,4 +272,51 @@
     binary_link_dir => '/usr/bin',
   }
 
+  if $majori < 9 {
+    # Registers orbd using update-alternatives
+    java::alternative { "java-alternatives-orbd-${vendor}-${version}-${arch}":
+      defaultJava     => $defaultJava,
+      vendor          => $vendor,
+      version         => $version,
+      arch            => $arch,
+      binary_name     => 'orbd',
+      binary_dir      => "${installDir}/bin",
+      binary_link_dir => '/usr/bin',
+    }
+
+    # Registers servertool using update-alternatives
+    java::alternative { "java-alternatives-servertool-${vendor}-${version}-${arch}":
+      defaultJava     => $defaultJava,
+      vendor          => $vendor,
+      version         => $version,
+      arch            => $arch,
+      binary_name     => 'servertool',
+      binary_dir      => "${installDir}/bin",
+      binary_link_dir => '/usr/bin',
+    }
+
+    # Registers tnameserv using update-alternatives
+    java::alternative { "java-alternatives-tnameserv-${vendor}-${version}-${arch}":
+      defaultJava     => $defaultJava,
+      vendor          => $vendor,
+      version         => $version,
+      arch            => $arch,
+      binary_name     => 'tnameserv',
+      binary_dir      => "${installDir}/bin",
+      binary_link_dir => '/usr/bin',
+    }
+
+    # Registers jhat using update-alternatives
+    java::alternative { "java-alternatives-jhat-${vendor}-${version}-${arch}":
+      defaultJava     => $defaultJava,
+      vendor          => $vendor,
+      version         => $version,
+      arch            => $arch,
+      binary_name     => 'jhat',
+      binary_dir      => "${installDir}/bin",
+      binary_link_dir => '/usr/bin',
+    }
+
+  }
```

Comment:
don't you think we should declare the new tools provided with jdk 9 ?

https://subscription.packtpub.com/book/application_development/9781786461407/1/01lvl1sec14/using-new-tools-in-jdk-9

Reply:
good catch. All new tools are now declared